src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -107,17 +101,13 @@ public class BlobFileStore : IFileStore
             BlockBlobClient blob = GetInstanceBlockBlob(versionedInstanceIdentifier);
 
             Stream stream = null;
-            StorageTransferOptions storageTransferOptions = new StorageTransferOptions
-            {
-                MaximumConcurrency = _blobDataStoreConfiguration.RequestOptions.DownloadMaximumConcurrency,
-            };
+            var blobOpenReadOptions = new BlobOpenReadOptions(allowModifications: false);
 
             await ExecuteAsync(async () =>
             {
-                stream = _recyclableMemoryStreamManager.GetStream(GetFileStreamTagName);
-                await blob.DownloadToAsync(stream, conditions: null, storageTransferOptions, cancellationToken);
+                stream = await blob.OpenReadAsync(blobOpenReadOptions, cancellationToken);
```

Comment:
I took a look at the underlying `LazyLoadingReadOnlyStream` that is returned here, and the buffer it uses to hold chunks of the blob are rented from the shared `ArrayPool<byte>`. So I think this will be an overall win!